extensions/amp-bind/0.1/bind-impl.js

```diff
@@ -912,6 +938,48 @@ export class Bind {
   }
 
   /**
+   * Deep merge object b into object a. Both a and b can only contain
+   * primitives, arrays, and objects created by Object.create(null).
+   * Arrays are replaced, not merged. Other objects are recursively merged.
+   * @param {!Object} a the destination object
+   * @param {!Object} b
+   * @param {number=} opt_depth The depth of the objects being merged
+   * @return {!Object}
+   */
+  deepMerge_(a, b, opt_depth) {
+    const depth = opt_depth || 0;
+    if (depth > this.maxMergeDepth_) {
+      user().error(TAG, 'merge depth exeeds limit');
+      return a;
+    }
+    Object.keys(b).forEach(key => {
+      const newValue = b[key];
+      // Perform a deep merge IFF
+      // 1: Both a and b have the same property
+      if (Object.hasOwnProperty.call(a, key)) {
+        const oldValue = a[key];
+        // 2: AND the properties on both a and b are non-null objects
+        if (Object(newValue) === newValue && Object(oldValue) == oldValue) {
+          // 3: AND the properties on both a and b are not arrays
+          // (we just want to overwrite arrays)
+          if (!Array.isArray(newValue) && !Array.isArray(oldValue)) {
```

Comment:
You can use `Object.prototype.toString.call(newValue) === [object Object]` instead. This will also constrain deepMerge_ to only plain objects instead of all non-Array objects.